frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java

```diff
@@ -270,46 +275,82 @@ public void queueRepositoryUpdateAndModelRefresh(@UI Runnable doOnUIThreadWhenRe
 
     @UIEffect
-    GitMacheteGraphTableMouseAdapter(EnhancedGraphTable outer) {
-      this.outer = outer;
+    EnhancedGraphTableMouseAdapter(EnhancedGraphTable graphTable) {
+      this.graphTable = graphTable;
+      this.popupMenuListener = new EnhancedGraphTablePopupMenuListener(graphTable);
     }
 
     @Override
     @UIEffect
     public void mouseClicked(MouseEvent e) {
       Point point = e.getPoint();
-      int row = outer.rowAtPoint(point);
-      int col = outer.columnAtPoint(point);
+      int row = graphTable.rowAtPoint(point);
+      int col = graphTable.columnAtPoint(point);
 
       // check if we click on one of branches
       if (row < 0 || col < 0) {
         return;
       }
 
-      BranchOrCommitCell cell = (BranchOrCommitCell) outer.getModel().getValueAt(row, col);
+      BranchOrCommitCell cell = (BranchOrCommitCell) graphTable.getModel().getValueAt(row, col);
       IGraphItem graphItem = cell.getGraphItem();
       if (!graphItem.isBranchItem()) {
         return;
       }
 
-      outer.selectedBranchName = graphItem.asBranchItem().getBranch().getName();
+      graphTable.selectedBranchName = graphItem.asBranchItem().getBranch().getName();
 
       ActionManager actionManager = ActionManager.getInstance();
       if (SwingUtilities.isRightMouseButton(e)) {
         ActionGroup contextMenuActionGroup = (ActionGroup) actionManager.getAction(ActionGroupIds.ACTION_GROUP_CONTEXT_MENU);
         var actionPopupMenu = actionManager.createActionPopupMenu(ACTION_PLACE_CONTEXT_MENU, contextMenuActionGroup);
-        actionPopupMenu.getComponent().show(outer, (int) point.getX(), (int) point.getY());
+        JPopupMenu popupMenu = actionPopupMenu.getComponent();
+        popupMenu.addPopupMenuListener(popupMenuListener);
+        popupMenu.show(graphTable, (int) point.getX(), (int) point.getY());
       } else if (SwingUtilities.isLeftMouseButton(e) && e.getClickCount() == 2 && !e.isConsumed()) {
         e.consume();
-        DataContext dataContext = DataManager.getInstance().getDataContext(outer);
+        DataContext dataContext = DataManager.getInstance().getDataContext(graphTable);
         var actionEvent = AnActionEvent.createFromDataContext(ACTION_PLACE_CONTEXT_MENU, new Presentation(), dataContext);
         actionManager.getAction(ACTION_CHECK_OUT).actionPerformed(actionEvent);
       }
     }
   }
 
+  private static class EnhancedGraphTablePopupMenuListener implements PopupMenuListener {
+    private final EnhancedGraphTable graphTable;
+
+    @UIEffect
+    EnhancedGraphTablePopupMenuListener(EnhancedGraphTable graphTable) {
+      this.graphTable = graphTable;
+    }
+
+    @Override
+    @UIEffect
+    public void popupMenuWillBecomeVisible(PopupMenuEvent popupMenuEvent) {
+      // This delay is needed to avoid `focus transfer` effect when at the beginning row selection is light blue
+      // but when context menu is created (in fraction of second) became dark blue/gray because of focus loose.
```

Comment:
```suggestion
      // This delay is needed to avoid `focus transfer` effect when at the beginning row selection is dark blue,
      // but when context menu is created (in a fraction of second), selection loses focus to the context menu and becomes gray.
```
(not sure about the colors though; this was my impression)

Reply:
I'd stay with light and dark blue ;)

Reply:
But yes, it can depend on the LCD screen quite a lot